### PR TITLE
Add Symfony 7 support

### LINKED ISSUE #7413: Symfony 7 compatibililty
# Problem description

Bring Symfony 7 compatibility to Sulu 2.6, but keep compatibility to Symfony 5 and 6.

Projects:

 - [x] sulu/sulu (running): https://github.com/sulu/sulu/pull/7156
 - [x] sulu/skeleton (running): https://github.com/sulu/skeleton/pull/246
 - [x] sulu/messenger: https://github.com/sulu/messenger/pull/6
 - [x] sulu/SuluContentBundle https://github.com/sulu/SuluContentBundle/pull/258
 - [x] sulu/SuluThemeBundle https://github.com/sulu/SuluThemeBundle/pull/35
     - [x] sylius/SyliusThemeBundle: https://github.com/Sylius/SyliusThemeBundle/pull/131
     - [x] sylius/SyliusThemeBundle: https://github.com/Sylius/SyliusThemeBundle/pull/134
 - [x] sulu/SuluArticleBundle https://github.com/sulu/SuluArticleBundle/pull/684
    - [x] Requires HeadlessBundle
    - [x] Requires AutomationBundle
 - [x] sulu/SuluAutomationBundle https://github.com/sulu/SuluAutomationBundle/pull/75
    - [x] https://github.com/php-task/php-task/ https://github.com/php-task/php-task/pull/49
    - [x] https://github.com/php-task/taskbundle https://github.com/php-task/TaskBundle/pull/55
 - [x] sulu/SuluHeadlessBundle https://github.com/sulu/SuluHeadlessBundle/pull/139
 - [x] sulu/SuluRedirectBundle  https://github.com/sulu/SuluRedirectBundle/pull/92
 - [x] sulu/SuluFormBundle https://github.com/sulu/SuluFormBundle/pull/390
 - [x] sulu/SuluCommentBundle https://github.com/sulu/SuluCommentBundle/pull/53
 - [x] sulu/SuluCommunityBundle https://github.com/sulu/SuluCommunityBundle/pull/172
 - [x] sulu/SuluSyliusConsumerBundle https://github.com/sulu/SuluSyliusConsumerBundle/pull/133
 - ~~[sulu/SuluSyliusProducerPlugin](https://github.com/sulu/SuluSyliusProducerPlugin)~~ This is related to the Sylius eco System: https://github.com/Sylius/Sylius/issues/15481 so currently we can not tackle it.

Add compatibility to Symfony 7, the bundles should still be compatible to PHP 8.0 and Sulu 2.5.

# Proposed solution

Mostly upgrade and see what is failing and check such things where changed in Symfony 7 support for Sulu. Mostly bundles need also add support for doctrine/persistence 3 before they can upgrade to Symfony 6.4 / 7.0.

Compatibility Sulu Core:

 - [x] doctrine-bridge 6.3 with validator 7.0: `Testing PHP Fatal error:  Declaration of Symfony\Bridge\Doctrine\Validator\DoctrineInitializer::initialize(object $object) must be compatible with Symfony\Component\Validator\ObjectInitializerInterface::initialize(object $object): void in /home/runner/work/sulu/sulu/vendor/symfony/doctrine-bridge/Validator/DoctrineInitializer.php on line 34` ( https://github.com/symfony/symfony/pull/51630 )
 - [x] sulu SecurityBundle and security-http 7.0 `PHP Fatal error:  Declaration of Sulu\Bundle\SecurityBundle\Security\AuthenticationEntryPoint::start(Symfony\Component\HttpFoundation\Request $request, ?Symfony\Component\Security\Core\Exception\AuthenticationException $authException = null) must be compatible with Symfony\Component\Security\Http\EntryPoint\AuthenticationEntryPointInterface::start(Symfony\Component\HttpFoundation\Request $request, ?Symfony\Component\Security\Core\Exception\AuthenticationException $authException = null): Symfony\Component\HttpFoundation\Response in /home/runner/work/sulu/sulu/src/Sulu/Bundle/SecurityBundle/Security/AuthenticationEntryPoint.php on line 25` 
 - [x] doctrine/persistence 3.0 support
    - [x] Fix deprecations https://github.com/sulu/sulu/pull/7134
    - [x] Fix merge deprecations in fixtures https://github.com/sulu/sulu/issues/7184
    - [x] Allow doctrine/persistence 3.0 support https://github.com/sulu/sulu/pull/6643
    - [x] https://github.com/sulu/sulu/issues/7198
 - [x] Declaration of Sulu\Component\Content\Form\DataTransformer\DocumentToUuidTransformer::transform($document) must be compatible with Symfony\Component\Form\DataTransformerInterface::transform(mixed $value): mixed in /home/runner/work/sulu/sulu/src/Sulu/Component/Content/Form/DataTransformer/DocumentToUuidTransformer.php on line 32 https://github.com/sulu/sulu/pull/7190
 - [x] https://github.com/sulu/sulu/pull/7347
    - [x] VoterInterface changes
    - [x] UserProviderInterface changes
    - [x] Security Class changes
 - [x] https://github.com/sulu/sulu/pull/7357
    - [x] ContainerAwareInterface removal
 - [x] https://github.com/sulu/sulu/pull/7359
    - [x] LoaderInterface changes
    - [x] ExpressionLanguageProvider changes 
 - [x] https://github.com/sulu/sulu/pull/7371
    - [x] Add compatibility to Symfony 7 to ActivateResolveTargetEntityResolverPas
 - [x] https://github.com/sulu/sulu/pull/7368
    - [x] Add Symfony 7 compatibility to FieldDescriptorFactory
 - [x] https://github.com/sulu/sulu/pull/7369
    - [x] Fix PreviewKernel and RefreshCommand compatibility
 - [x] https://github.com/sulu/sulu/pull/7374
    - [x] Refactor doctrine.event_subscriber to doctrine.event_listener for Symfony 7 compatibility
 - [x] https://github.com/sulu/sulu/pull/7380
    - [x] Refactor JMS/Serializer annotations to attributes
 - [x] https://github.com/sulu/sulu/pull/7382
    - [x] Refactor ParameterBag in InputBag in Core Tests
 - [x] https://github.com/sulu/sulu/pull/7390
    - [x] Refactor HeaderBag to InputBag in Core Tests
 - [x] https://github.com/sulu/sulu/pull/7392
    - [x] Fix some return types in tests expections

Dependencies:

 - [x] friendsofsymfony/http-cache-bundle
    - [x] https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/pull/613
    - [x] 3.x release
    - [x] Sulu compatibility: https://github.com/sulu/sulu/pull/7452
 - [x] friendsofsymfony/http-cache
    - [x] https://github.com/FriendsOfSymfony/FOSHttpCache/pull/552
    - [x] 3.x release
 - [x] dantleech/phpcr-migrations-bundle
    - [x] dantleech/phpcr-migrations (https://github.com/phpcr/phpcr-migrations/pull/18)
    - [x] requires doctrine/phpcr-bundle
    - [x] requires symfony-cmf/testing
 - [x] doctrine/phpcr-bundle
    - [x] https://github.com/doctrine/DoctrinePHPCRBundle/pull/387
 - [x] doctrine/doctrine-fixtures-bundle (Requires 4.0: https://github.com/doctrine/DoctrineFixturesBundle/pull/395)
    - [x] https://github.com/sulu/sulu/pull/7161
 - [x] friendsofsymfony/rest-bundle
    - [x] https://github.com/FriendsOfSymfony/FOSRestBundle/pull/2400
 - [x] friendsofsymfony/jsrouting-bundle
    - [x] https://github.com/FriendsOfSymfony/FOSJsRoutingBundle/pull/471
 - [x] handcraftedinthealps/rest-routing-bundle
    - [x] requires friendsofsymfony/rest-bundle
 - [x] jms/serializer-bundle
    - [x] https://github.com/schmittjoh/JMSSerializerBundle/pull/937
 - [x] massive/build-bundle
    - [x] requires matthiasnoback/symfony-dependency-injection-test Symfony 7 support
    - [x] https://github.com/massiveart/MassiveBuildBundle/pull/25
 - [x] massive/search-bundle
    - [x] requires matthiasnoback/symfony-dependency-injection-test Symfony 7 support
    - [x] https://github.com/massiveart/MassiveSearchBundle/pull/172
 - [x] symfony-cmf/routing-bundle
    - [x] https://github.com/symfony-cmf/routing-bundle/pull/488
 - [x] matthiasnoback/symfony-config-test https://github.com/SymfonyTest/SymfonyDependencyInjectionTest/releases/tag/5.0.0
 - [x] matthiasnoback/symfony-dependency-injection-test https://github.com/SymfonyTest/SymfonyDependencyInjectionTest/releases/tag/5.0.0
 - [x] phpcr/phpcr-shell https://github.com/phpcr/phpcr-shell/pull/224
     - [x] https://github.com/phpcr/phpcr-utils/pull/214 (release missing)
     - [x] https://github.com/phpspec/phpspec/pull/1452
     - [x] https://github.com/Behat/Behat/pull/1442
 - [x] scheb/2fa-bundle
     - [x] Allow 7.x version https://github.com/sulu/sulu/pull/7254
     - [x] own 7.x version: https://github.com/scheb/2fa/blob/7.x/composer.json
 - [x] toflar/psr6-symfony-http-cache-store
    - [x] https://github.com/Toflar/psr6-symfony-http-cache-store/pull/46
 - [x] stof/doctrine-extensions-bundle
    - [x] requires first gedmo/extension https://github.com/doctrine-extensions/DoctrineExtensions

### PR DESCRIPTION
**Pull request by TheCadien**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes [#7413](https://github.com/sulu/sulu/issues/7413)| License | MIT

#### What's in this PR?

Allow Symfony 7 Packages.

part of https://github.com/sulu/sulu/issues/7413

### CODE REVIEW

File: .php-cs-fixer.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...nd_double_colon":true,"no_spaces_after_function_name":true,"no_trailing_whitespace":true,"no_trailing_whitespace_in_comment":true,"single_blank_line_at_eof":true,"single_line_after_imports":true,"spaces_inside_parentheses":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true,"encoding":true,"full_opening_tag":true,"header_comment":{"header":"This file is part of Sulu.\n\n(c) Sulu GmbH\n\nThis source file is subject to the MIT license that is bundled\nwith this source code in the file LICENSE."}},"hashes":{"\/private\/var\/folders\/02\/38pv02ln53sf66jz76rj189m0000gn\/T\/PHP CS Fixertemp_folder415\/DependencyInjection\/Configuration.php":"fc7a93d87c407e10fcb2611bd689efe4","\/private\/var\/folders\/02\/38pv02ln53sf66jz76rj189m0000gn\/T\/PHP CS Fixertemp_folder121\/DependencyInjection\/Configuration.php":"fc7a93d87c407e10fcb2611bd689efe4","\/private\/var\/folders\/02\/38pv02ln53sf66jz76rj189m0000gn\/T\/PHP CS Fixertemp_folder1303\/DependencyInjection\/Configuration.php":"c3ce9e4dc7c15c60b05bed438bec7d8f","\/private\/var\/folders\/02\/38pv02ln53sf66jz76rj189m0000gn\/T\/PHP CS Fixertemp_folder598\/DependencyInjection\/Configuration.php":"c3ce9e4dc7c15c60b05bed438bec7d8f"}}
```
Comment (alexander-schranz) [inline .php-cs-fixer.cache]: should be part of gitignore

File: Resources/config/gone_subscriber.xml [hunk 2]
```diff
@@ -19,7 +19,7 @@
                  class="Sulu\Bundle\RedirectBundle\GoneSubscriber\GoneEntitySubscriber">
             <argument type="service" id="sulu_redirect.redirect_route_manager"/>
 
-            <tag name="doctrine.event_subscriber"/>
+            <tag name="doctrine.event_listener" event="LifecycleEventArgs"/>
```
Comment (alexander-schranz) [inline Resources/config/gone_subscriber.xml]: think event should be `preRemove`

File: composer.json [hunk 3]
```diff
@@ -7,18 +7,18 @@
         "php": "^7.2 || ^8.0",
         "sulu/sulu": "^2.2.5 || ^2.3@dev",
         "ramsey/uuid": "^3.1 || ^4.0",
-        "symfony/dependency-injection": "^4.3 || ^5.0 || ^6.0",
-        "symfony/config": "^4.3 || ^5.0 || ^6.0",
-        "symfony/console": "^4.3 || ^5.0 || ^6.0",
-        "symfony/http-foundation": "^4.3 || ^5.0 || ^6.0",
+        "symfony/dependency-injection": "^4.3 || ^5.0 || ^6.0 || ^6.0",
```
Comment (alexander-schranz) [inline composer.json]: 👀

File: composer.json [hunk 4]
```diff
@@ -30,12 +30,12 @@
         "phpstan/phpstan-symfony": "^1.0",
         "phpunit/phpunit": "^8.0",
         "php-cs-fixer/shim": "^3.0",
-        "symfony/browser-kit": "^4.3 || ^5.0 || ^6.0",
-        "symfony/dotenv": "^4.3 || ^5.0 || ^6.0",
+        "symfony/browser-kit": "^4.3 || ^5.0 || ^6.0 || ^6.0",
+        "symfony/dotenv": "^4.3 || ^5.0 || ^6.0 || ^6.0",
         "symfony/monolog-bundle": "^3.1",
         "jackalope/jackalope-doctrine-dbal": "^1.3.4",
```
Comment (alexander-schranz) [inline composer.json]: this needs also increased:

```suggestion
        "jackalope/jackalope-doctrine-dbal": "^1.3.4 || ^2.0",
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: composer.json (hunk 3 above)
Comment (TheCadien) [reply composer.json]: uff lol

File: Tests/Unit/Controller/RedirectControllerTest.php [hunk 5]
```diff
@@ -45,19 +46,17 @@ protected function setUp(): void
         $this->controller = new WebsiteRedirectController();
 
         $this->request = $this->prophesize(Request::class);
-        $this->queryBag = $this->prophesize(ParameterBag::class);
+        $this->queryBag = class_exists(InputBag::class) ? new InputBag() : new ParameterBag();
```
Comment (alexander-schranz) [inline Tests/Unit/Controller/RedirectControllerTest.php]: when already refctoring this to no mock I would go with not mocking the whole Request object.

```
$this->request = Request::create();
```

then you can easily use:

```
$request->query->set('test', 1);
$request->query->set('my-parameter', 'awesome sulu');
```

This avoids have to bridge between parameterbag and inputbag

File: Tests/Unit/Controller/RedirectControllerTest.php [hunk 6]
```diff
@@ -45,22 +40,22 @@ protected function setUp(): void
...
-        $this->request->reveal()->query = $this->queryBag;
     }
 
     public function testRedirect()
     {
         $target = '/test';
         $statusCode = 301;
 
+        $request = Request::create($target);
+        $request->query = $this->queryBag;
```
Comment (alexander-schranz) [inline Tests/Unit/Controller/RedirectControllerTest.php]: would call `$request->query->set` and not create `this->queryBag` longer.

Then the whole class_exists inputbag vs parameterbag part can be removed

Comment (TheCadien) [reply Tests/Unit/Controller/RedirectControllerTest.php]: ahhh now i see what you mean :D !

**Pull request merged**